dask_expr/_expr.py

```diff
@@ -2867,7 +2867,7 @@ def are_co_aligned(*exprs, allow_broadcast=True):
     ancestors = [set(non_blockwise_ancestors(e)) for e in exprs]
     unique_ancestors = {
         # Account for column projection within IO expressions
-        _tokenize_partial(item, ["columns", "_series"])
+        _tokenize_partial(item, ["columns", "_series", "_dataset_info_cache"])
```

Comment:
See https://github.com/dask-contrib/dask-expr/issues/907 for the root cause discussion. What I'm doing here is a pretty brittle patch imo